src/alert/alert.js

```diff
@@ -19,12 +19,13 @@ angular.module('ui.bootstrap.alert', [])
   };
 })
 
-.directive('dismissOnTimeout', ['$timeout', function($timeout) {
+.directive('dismissOnTimeout', ['$interval', function($interval) {
   return {
     require: 'alert',
     link: function(scope, element, attrs, alertCtrl) {
-      $timeout(function(){
+      var intervalCanceller = $interval(function() {
         alertCtrl.close();
+        $interval.cancel(intervalCanceller);
```

Comment:
not sure how this works with protractor, but it should be enough to set count to 1:
```js
$interval(function() {
     alertCtrl.close();
}, parseInt(attrs.dismissOnTimeout, 10), 1);
```